pkg/kubelet/resources.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+func (kl *Kubelet) defaultPodLimitsForDownwardApi(pod *api.Pod) (*api.Pod, error) {
+	capacity := make(api.ResourceList)
+	lastUpdatedNodeObject := kl.lastUpdatedNodeObject.Load()
+	if lastUpdatedNodeObject == nil {
+		glog.Errorf("Failed to find node object in cache. Expected a non-nil object in the cache.")
```

Comment:
AFAIK not possible today. Just defensive programming. In case someone changes the order of initialization in the kubelet, which has happened a lot of times in the past, this error might be triggered.
In retrospect, it might make sense to return error instead of just logging.